source/MaterialXView/Viewer.cpp

```diff
@@ -1501,6 +1503,16 @@ void Viewer::loadDocument(const mx::FilePath& filename, mx::DocumentPtr librarie
                 if (materialNode)
                 {
                     std::string udim = mat->getUdim();
+                    if (udim.empty())
```

Comment:
I guess this is okay. It's a bit hidden what it's doing and any material with 4-digit numbers in the name will be treated as a udim mapping. Should it be an advanced option to allow turning this on / off ?

Reply:
I've added comments to help clarify the new proposed behavior, and let me know how this looks to you.

Reply:
As far as I could see, UDIM syntax at the Document and at the Material level doesn't seem to be documented at all in the specification, so I'm wondering if this is just a good opportunity to improve this vs code comments here ?

Reply:
@kwokcb It's beyond the scope of this PR, but I'd agree documentation of the expected behavior for UDIMs is an important goal for MaterialX.

In particular, the AOUSD Materials Working Group is expecting to contribute a normative description of UDIMs to the MaterialX specification, as historically the behavior of UDIMs has only been defined by industry conventions and product-specific documentation.